File: java/Mediator/src/Buyer.java


public class Buyer {
	Mediator mediator;
	String currency;

	public Buyer(Mediator mediator, String currency) {
		super();
		this.mediator = mediator;
		this.currency = currency;
		mediator.addBuyer(this); // blank 
	}

	public boolean attemptToPurchase(float bid) {
		System.out.println("Buyer attempting a bid of " + currency +" " + bid + " .");
		return mediator.placeBid(bid, currency); // blank
	}
	
}

File: ruby/Mediator/buyer.rb


class Buyer
  attr_accessor :mediator
  attr_accessor :currency

  def initialize(mediator,currency)
    @mediator = mediator
    @currency = currency
    mediator.addBuyer(self)
  end

  def attemptToPurchase(bid)
    print "Buyer attempting a bid of #{currency} #{bid}.\n"
    return mediator.placeBid(bid, currency)
  end
end